subsys/mpsl/cx/Kconfig

```diff
@@ -4,8 +4,20 @@
 # SPDX-License-Identifier: LicenseRef-Nordic-5-Clause
 #
 
+DT_COMPAT_CX_NRF700X := nordic,nrf700x-coex
+# config MPSL_CX_SUPPORTED
+
+config MPSL_CX_ANY_SUPPORT
+	bool
+	default $(dt_nodelabel_enabled,nrf_radio_coex)
+
+config MPSL_CX_NRF700X_SUPPORT
+	bool
+	default $(dt_nodelabel_has_compat,nrf_radio_coex,$(DT_COMPAT_CX_NRF700X))
+
 config MPSL_CX
 	bool "Radio Coexistence interface support"
+	default MPSL_CX_ANY_SUPPORT && WIFI && (BT || NRF_802154_RADIO_DRIVER || NRF_802154_SER_HOST)
```

Comment:
Is the use of default correct here?

As this is written I may enable MPSL_CX even though MPSL_CX_ANY_SUPPORT is false.

Reply:
Yes, the use of default is correct. The idea is to automatically enable coex if certain prerequisites are met, not disable it if they aren't.
The idea is that the user can add his own out of tree coex module. We don't want to force a way in which he should do it.

Reply:
According to https://github.com/nrfconnect/sdk-nrf/pull/8750/files#diff-aa6326f19f0b88a6282866c442a23fe0f5a977f7047cb0f7b687a691111d1c33R221

the intended semantics is for the device tree node to be required. But as this kconfig is written one can omit the device tree node and enable MPSL_CX_ANY_SUPPORT instead.

Either the docs should be updated to reflect that a DT node is not needed. Or we should have depends on MPSL_CX_ANY_SUPPORT.

Note that coex will still be automatically enabled.

Reply:
According to https://github.com/nrfconnect/sdk-nrf/pull/8750/files#diff-aa6326f19f0b88a6282866c442a23fe0f5a977f7047cb0f7b687a691111d1c33R221 the device tree node is required.

Either the docs should be updated to reflect that it is optional or MPSL_CX should use "depends on" for ANY_SUPPORT.

Note that coex will still be enabled automatically.

Reply:
Ok, done, now it depends on `ANY_SUPPORT`